File: src/main/webapp/views/notification/controllers/notificationController.js
(function () {
    angular
        .module("weProApp")
        .controller("NotificationController", notificationController);

    
    function notificationController(NotificationService, $routeParams) {
        var vm = this;
        vm.notificationList = undefined;
        vm.uid = $routeParams['uid'];
        /**
         * This method will load all the notification  for a user
         */
        function loadAllNotificationsForUser() {
        	NotificationService.getAllNotificationForUser(vm.uid)
                .then(function (projects) {
                	console.log(projects.data);
                	vm.notificationList = projects.data;
                	
                });
        }
        
        loadAllNotificationsForUser();
        
        /**
         * Method used to mark a notification as read
         */
        this.markRead = function markRead(notificationId) {
        	NotificationService.markNotificationAsRead(notificationId)
            .then(function (projects) {
            	console.log(projects.data);
            	vm.notificationList = projects.data;
            	
            });
          };
          
          /**
           * Method used to delete a user notification
           */
          this.deleteNotification = function deleteNotification(notificationId) {
          	NotificationService.deleteNotification(notificationId)
              .then(function (projects) {
              	console.log(projects.data);
              	vm.notificationList = projects.data;
              	
              });
            };
    }
})();

File: src/main/java/com/ip/wePro/Notification/NotificationController.java
package com.ip.wePro.Notification;

import java.util.ArrayList;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.ip.wePro.User.User;
import com.ip.wePro.skills.Skills;

@RestController
@RequestMapping("/notification")
public class NotificationController {
	
	@Autowired
	NotificationService notificationService;
	
	/**
	 * To Test send email notification part through rest api
	 */
	/*@GetMapping("/get")
    public void getAllProjects() throws InterruptedException{
		
		Skills skills = new Skills();
		skills.setId(1);
		skills.setName("JAVA");
		List<Integer> list = new ArrayList<>();
		list.add(1);
        notificationService.sendNotificationForProject(list);
    }*/
	
	/**
	 * Method to pull out all the user notifications based on user id
	 * @param uid
	 * @return
	 */
	@GetMapping("/get/{uid}")
    public List<Notification> getAllUserNotifications(@PathVariable int uid){
		
        return notificationService.getAllUserNotifications(uid);
    }
	
	/**
	 * Method used to mark a notification as read
	 * @param notificationId
	 * @return
	 */
	@GetMapping("/markRead/{notificationId}")
    public List<Notification> markNotificationAsRead(@PathVariable Long notificationId){
		
        return notificationService.markNotificationAsRead(notificationId);
    }
	
	/**
	 * Method used to delete a user notification
	 * @param notificationId
	 * @return
	 */
	@GetMapping("/delete/{notificationId}")
    public List<Notification> deleteNotification(@PathVariable Long notificationId){
		
        return notificationService.deleteNotification(notificationId);
    }


}
